### PR TITLE
Generalized refactor

### PR DESCRIPTION
**Pull request by sarahsturgeon**
Tested each feature on both storage options at a simple level - no errors or misbehavior! 🎉

### CODE REVIEW

File: lua/cfc_time/server/storage_options/sqlite.lua [hunk 1]
```diff
@@ -136,7 +136,7 @@ function storage:UpdateBatch( batchData )
 end
 
 function storage:GetTotalTime( steamID, callback )
-    local data = storage:QueryTotalTime( steamID )
+    local data = self:QueryTotalTime( steamID )
     local sum = data[1]["SUM(duration)"]
```
Comment (samuelWilliams99) [inline lua/cfc_time/server/storage_options/sqlite.lua:140]: Should the whole `[1]["SUM(duration)"]` be done in `QueryTotalTime`? or do we want that to always return the raw query result

File: lua/cfc_time/server/time_tracking.lua [hunk 2]
```diff
@@ -105,21 +107,41 @@ function ctime:initPlayer( ply )
...
+        local initialTime = {
+            seconds = 0,
+
+            add = function( this, seconds )
+                this.seconds = this.seconds + seconds
+            end,
+
+            set = function( this, seconds )
+                this.seconds = seconds
+            end
```
Comment (samuelWilliams99) [inline lua/cfc_time/server/time_tracking.lua:122]: If we have `add` and `set`, should we not also have a `get` function?

**Review by samuelWilliams99 — APPROVED** [review]
Looks good, couple possible suggestions but not required or anything

File: lua/cfc_time/server/storage_options/sqlite.lua (hunk 1 above)
Comment (sarahsturgeon) [reply lua/cfc_time/server/storage_options/sqlite.lua:140]: Mmm, given that this is just used to serve the externally-facing `GetTotalTime` function, I'm thinking it's fine for it to return a dumb response. `QueryTotalTime`'s job is to query the total time, `GetTotalTime`'s job is to use that query result and extract meaning.

There is something to be said about whether or not `GetTotalTime` needs to know/care about the structure of the result, though..

What do you think?

File: lua/cfc_time/server/time_tracking.lua (hunk 2 above)
Comment (sarahsturgeon) [reply lua/cfc_time/server/time_tracking.lua:122]: okayyy

**Pull request merged**